### ISSUE TITLE
Twitter refresher raises "Bad Authentication data" exception

### ISSUE THREAD

**Issue by tulir** [feedbin/refresher#11]
I have a self-hosted Feedbin instance with all the extra stuff set up and for the most part it works fine. Everything with RSS feeds works fine, authorizing feedbin to access twitter works fine and subscribing to a feed makes the few latest tweets appear. However, a bit after subscribing to a twitter feed, the refresher will start raising `Twitter::Error::BadRequest: Bad Authentication data.` exceptions with irregular intervals (10 seconds - 5 minutes)

https://gist.github.com/tulir/9c570cb45526aa1f8c2bc0f2255c0434

How should I go about debugging the problem?

**Comment by benubois** [comment]
Hi!

I’m not familiar with this error. Are you running the latest version?

I’d try inspecting the requests the twitter library is making to see if it looks correct. You could configure it to connect through a proxy like [Charles](https://www.charlesproxy.com/).

**Comment by tulir** [comment]
It turned out I wasn't passing the twitter API key/secret environment variables to the refresher docker container. I also found out that feedkit uses `transform_values`, which doesn't exist in Ruby 2.3. After fixing the environment variables and switching to ruby 2.5, it seems to work fine.

Would be nice if the feedbin side services (refresher and image) had some setup instructions/environment requirements/etc. Especially the [image README](https://github.com/feedbin/image/blob/master/README.md) is a bit bare, though surprisingly I got it to work just fine.
